src/Shared/Recruit.Vacancies.Client/Infrastructure/Services/VacancySummariesProvider/VacancySummariesProvider.cs

```diff
@@ -0,0 +1,81 @@
+using Esfa.Recruit.Vacancies.Client.Domain.Entities;
+using Esfa.Recruit.Vacancies.Client.Infrastructure.Mongo;
+using Esfa.Recruit.Vacancies.Client.Infrastructure.QueryStore.Projections;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using MongoDB.Bson;
+using MongoDB.Driver;
+using Polly;
+
+namespace Esfa.Recruit.Vacancies.Client.Infrastructure.Services.VacancySummariesProvider
+{
+    internal sealed class VacancySummariesProvider : MongoDbCollectionBase, IVacancySummariesProvider
+    {
+        private const string PathSpecifierFieldName = "path";
+
+        private const string PreserveNullAndEmptyArraysSpecifierFieldName = "preserveNullAndEmptyArrays";
+
+        public VacancySummariesProvider(ILoggerFactory loggerFactory, IOptions<MongoDbConnectionDetails> details)
+            : base(loggerFactory, MongoDbNames.RecruitDb, MongoDbCollectionNames.Vacancies, details)
+        {
+        }
+
+        public async Task<IList<VacancySummary>> GetEmployerOwnedVacancySummariesByEmployerAccountAsync(string employerAccountId)
+        {
+            var match = new BsonDocument
+            {
+                {
+                    "$match",
+                    new BsonDocument
+                    {
+                        { "employerAccountId", employerAccountId },
+                        { "ownerType", OwnerType.Employer.ToString() },
+                        { "isDeleted", false }
+                    }
+                }
+            };
+
+            var aggPipeline = VacancySummaryAggQueryBuilder.GetAggregateQueryPipeline(match);
+
+            return await RunAggPipelineQuery(aggPipeline);
+        }
+
+        public async Task<IList<VacancySummary>> GetProviderOwnedVacancySummariesByUkprnAsync(long ukprn)
+        {
+            var match = new BsonDocument
+            {
+                {
+                    "$match",
+                    new BsonDocument
+                    {
+                        { "trainingProvider.ukprn", ukprn },
+                        { "ownerType", OwnerType.Provider.ToString() },
+                        { "isDeleted", false }
+                    }
+                }
+            };
+
+            var aggPipeline = VacancySummaryAggQueryBuilder.GetAggregateQueryPipeline(match);
+
+            return await RunAggPipelineQuery(aggPipeline);
+        }
+
+        private async Task<IList<VacancySummary>> RunAggPipelineQuery(BsonDocument[] pipeline)
+        {
+            var mongoQuery = pipeline.ToJson();
+
+            var db = GetDatabase();
+            var collection = db.GetCollection<BsonDocument>(MongoDbCollectionNames.Vacancies);
+
+            var aggResults = await RetryPolicy.ExecuteAsync(async context => await collection.AggregateAsync<VacancySummaryAggQueryResponseDto>(pipeline),
+                                                            new Context(nameof(RunAggPipelineQuery)));
+
+            var vacancySummaries = await aggResults.ToListAsync();
```

Comment:
The `ToListAsync()` needs to be inside the retry as this is the bit that does the fetching from the DB so can we get cursor errors here.  I'm updating other instances of this as part of ER-898.

Reply:
👍 in terms of debugging or rewriting, don't worry about it. @mgwilliam wrote it originally and he is good at Congo queries. In terms of performance against Cosmos, will have to test this but I am confident we will be hitting the db less number of times.